cmd/doracled/cmd/get_oracle_key.go

```diff
@@ -16,69 +18,96 @@ import (
 	tos "github.com/tendermint/tendermint/libs/os"
 )
 
-var getOracleKeyCmd = &cobra.Command{
-	Use:   "get-oracle-key",
-	Short: "Get a shared oracle private key",
-	Long: `Get a shared oracle private key from Panacea.
+func getOracleKeyCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "get-oracle-key",
+		Short: "Get a shared oracle private key",
+		Long: `Get a shared oracle private key from Panacea.
 The encrypted oracle private key can only be decrypted using the node key, which is generated in SGX-enabled environment.
 The node key must be the same with the one stored in OracleRegistration KV store in Panacea.
 After decrypted, the oracle private key is sealed and stored as a file named oracle_priv_key.sealed securely.
 This oracle private key can also be accessed in SGX-enabled environment using the promised binary.
 `,
-	RunE: func(cmd *cobra.Command, args []string) error {
-		// if there is a node key already, return error
-		if !tos.FileExists(nodePrivKeyPath) {
-			return errors.New("no node_priv_key.sealed file")
-		}
-
-		// get config
-		conf, err := config.ReadConfigTOML(getConfigPath())
-		if err != nil {
-			return fmt.Errorf("failed to read config.toml: %w", err)
-		}
-
-		// get existing node key
-		nodePrivKeyBz, err := sgx.UnsealFromFile(nodePrivKeyPath)
-		if err != nil {
-			return fmt.Errorf("failed to unseal node_priv_key.sealed file: %w", err)
-		}
-		nodePrivKey, nodePubKey := crypto.PrivKeyFromBytes(nodePrivKeyBz)
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// if there is a node key already, return error
+			if !tos.FileExists(nodePrivKeyPath) {
+				return errors.New("no node_priv_key.sealed file")
+			}
+
+			// get config
+			conf, err := config.ReadConfigTOML(getConfigPath())
+			if err != nil {
+				return fmt.Errorf("failed to read config.toml: %w", err)
+			}
+
+			// get existing node key
+			nodePrivKeyBz, err := sgx.UnsealFromFile(nodePrivKeyPath)
+			if err != nil {
+				return fmt.Errorf("failed to unseal node_priv_key.sealed file: %w", err)
+			}
+			nodePrivKey, nodePubKey := crypto.PrivKeyFromBytes(nodePrivKeyBz)
+
+			// get unique ID
+			selfEnclaveInfo, err := sgx.GetSelfEnclaveInfo()
+			if err != nil {
+				return fmt.Errorf("failed to get self enclave info: %w", err)
+			}
+			uniqueID := base64.StdEncoding.EncodeToString(selfEnclaveInfo.UniqueID)
+
+			// get oracle account from mnemonic.
+			oracleAccount, err := getOracleAccount(cmd, conf.OracleMnemonic)
+			if err != nil {
+				return fmt.Errorf("failed to get oracle account from mnemonic: %w", err)
+			}
+
+			trustedBlockInfo, err := readTrustedBlockInfoFile()
+			if err != nil {
+				return fmt.Errorf("failed to read trusted block info file: %w", err)
+			}
+			fmt.Printf("%v will be used in query client", trustedBlockInfo)
```

Comment:
There doesn't seem to be anywhere that uses trusted BlockInfo, where do we use it?

Reply:
Oh, It will use in the future. :)